### PR TITLE
Rewind feature for express-cli devnet, POS-1080

### PR DESCRIPTION
**Pull request by nikhil0360**
# Description

added a function for testing rewind in devnet blockchain. 

`../../bin/express-cli --rewind [numberOfBlocks]`, by default numberOfBlocks is 100. This will rewind the chain by the specified number of blocks. If numberOfBlocks > 128, it will rewind the chain by 128 blocks.

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [x] New test case for remote devnet

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have added at least 2 reviewer or the whole pos-v1 team
- [x] I have added sufficient documentation in code
- [x] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

# Cross repository changes

#### It should never be the case...

- [ ] This PR requires changes to bor
  - In case link the PR here:
- [ ] This PR requires changes to heimdall
  - In case link the PR here:

## Testing

- [ ] I have tested this code manually on local environment
- [x] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai

### Manual tests

Please complete this section with the steps you performed if you ran manual tests for this functionality, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: src/express-cli.js [hunk 1]
```diff
@@ -304,5 +306,19 @@ export async function cli() {
...
+  } else if (options.rewind) {
+    console.log('📍Command --rewind')
+    if (!checkDir(false)) {
+      console.log(
+        '❌ The command is not called from the appropriate devnet directory!'
+      )
+      process.exit(1)
+    }
+    if (options.rewind === true) {
+      options.rewind = 100
```
Comment (marcello33) [inline src/express-cli.js:318]: Will this override the number provided by the user?

File: src/express/commands/rewind.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  runSshCommand,
+  runSshCommandWithReturn,
+  maxRetries
+} = require('../common/remote-worker')
+
+export async function rewind(num) {
+  //   num = number of blocks to rewind
+  if (num > 128) {
+    console.log(
+      '📍number of blocks to rewind should should be less than equal to 128, setting to 128'
```
Comment (marcello33) [inline src/express/commands/rewind.js:12]: "should" is written twice. Also, can you leave an empty space before "number" and use capital N?

File: src/express/commands/rewind.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  const devnetType =
+    process.env.TF_VAR_DOCKERIZED === 'yes' ? 'docker' : 'remote'
+
+  const doc = await loadDevnetConfig(devnetType)
+
+  const borUsers = splitToArray(doc.devnetBorUsers.toString())
+  const borHosts = splitToArray(doc.devnetBorHosts.toString())
+
+  if (doc.devnetBorHosts.length > 0) {
+    console.log('📍Monitoring the first node', doc.devnetBorHosts[0])
```
Comment (marcello33) [inline src/express/commands/rewind.js:27]: Spaces here too

File: src/express/commands/rewind.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  } else {
+    console.log('📍No nodes to monitor, please check your configs! Exiting...')
+    process.exit(1)
+  }
+
+  const ip = `${borUsers[0]}@${borHosts[0]}`
+
+  const getBlockNumberCommand =
+    '/home/ubuntu/go/bin/bor attach ~/.bor/data/bor.ipc --exec "eth.blockNumber"'
```
Comment (marcello33) [inline src/express/commands/rewind.js:36]: Can you use `~/` instead of `/home/ubuntu/` to be user agnostic?

File: src/express/commands/rewind.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  const intitalBlockNumber = await runSshCommandWithReturn(
+    ip,
+    getBlockNumberCommand,
+    maxRetries
+  )
+  console.log(
+    `⏪ rewinding chain by ${num} blocks, \n⛓ current block number: ${intitalBlockNumber}`
```
Comment (marcello33) [inline src/express/commands/rewind.js:44]: Can you use the placeholder `📍` which visually recognises the `express-cli` logs?

File: src/express/commands/rewind.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  const restartCommand = 'sudo service bor restart'
+  await runSshCommand(ip, restartCommand, maxRetries)
+
+  const rewindedBlockNumber = await runSshCommandWithReturn(
+    ip,
+    getBlockNumberCommand,
+    maxRetries
+  )
+  console.log(
+    `⏪ rewinded chain by ${
```
Comment (marcello33) [inline src/express/commands/rewind.js:59]: Same as before

Comment (marcello33) [reply src/express/commands/rewind.js:59]: And in other places too, going forward

File: src/express/common/remote-worker.js [hunk 7]
```diff
@@ -47,6 +47,42 @@ export async function runSshCommand(ip, command, retries) {
   }
 }
 
+export async function runSshCommandWithReturn(ip, command, retries) {
```
Comment (marcello33) [inline src/express/common/remote-worker.js:50]: Why is this needed?

Comment (0xsharma) [reply src/express/common/remote-worker.js:50]: We needed to have the stdout in a variable which is not possible by the legacy `runSshCommand`, hence this was introduced.

Comment (marcello33) [reply src/express/common/remote-worker.js:50]: Yeah sure, I meant more like why we need the stdout var to be returned in the first place.
Also, maybe this can be merged in the previous function and avoid duplications by using some sort of extracted util function

Comment (nikhil0360) [reply src/express/common/remote-worker.js:50]: This function is used for returning the value of `eth.getBlock(num)` which was not possible with the `runSshCommand`.

**Review by marcello33 — APPROVED** [review]

**Review by Raneet10 — APPROVED** [review]
